server/plugin/api.go

```diff
@@ -265,6 +265,20 @@ func (p *Plugin) handleCreateSubscription(w http.ResponseWriter, r *http.Request
 		return
 	}
 
+	if statusCode, channelAccessErr := p.CheckValidChannelForSubscription(body.ChannelID, mattermostUserID); channelAccessErr != nil {
+		p.API.LogError(constants.ErrorCreateSubscription, "Error", channelAccessErr.Error())
+
+		message := channelAccessErr.Error()
+		responseStatusCode := statusCode
```

Comment:
any reason for this new variable for statusCode? We can do all the below without this variable.

Reply:
we need to pass the default message/status code normally but in some cases, we need to pass a custom message and error code which is done inside `if` block below

Reply:
see this https://github.com/Brightscout/mattermost-plugin-azure-devops/pull/29#discussion_r1090494656

Reply:
yeah but i think you can directly update the statusCode variable itself without the need for responseStatusCode

Reply:
because it is in if so you cant that outside this if anyways  I think

Reply:
Okay will remove in the next PR

Reply:
sure